tests/unit/src/test/scala/tests/NewFilesLspSuite.scala

```diff
@@ -19,26 +19,84 @@ class NewFilesLspSuite extends BaseLspSuite("new-files") {
   override def initializationOptions: Option[InitializationOptions] =
     Some(InitializationOptions.Default.copy(inputBoxProvider = Some(true)))
 
-  check("new-worksheet")(
+  check("new-worksheet-picked")(
     Some("a/src/main/scala/"),
-    "worksheet",
-    Some("Foo"),
+    Right(worksheet),
+    Right("Foo"),
     "a/src/main/scala/Foo.worksheet.sc",
     ""
   )
 
-  check("new-ammonite-script")(
+  check("new-worksheet-name-provided")(
     Some("a/src/main/scala/"),
-    "ammonite",
-    Some("Foo"),
+    Left(worksheet),
+    Right("Foo"),
+    "a/src/main/scala/Foo.worksheet.sc",
+    ""
+  )
+
+  check("new-worksheet-fully-provided")(
+    Some("a/src/main/scala/"),
+    Left(worksheet),
+    Left("Foo"),
+    "a/src/main/scala/Foo.worksheet.sc",
+    ""
+  )
+
+  check("new-ammonite-script")(
+    directory = Some("a/src/main/scala/"),
+    fileType = Right(ammonite),
```

Comment:
Could we actually run the 3 cases in one check? Since the parameters are basically the same, we could just generate all the combinations `Right` `Right`, `Right` `Left` and `Left` `Left`

Reply:
I think possibly yea, but the first approach I took was to just have one check per file type, and then in the check create the file normally, and then also the other two ways, but then the check code became pretty hairy trying to get all the checked messages right in various places. So I actually then changed it to what I have now. It's more verbose, but I find it way easier to reason about.

Reply:
I'm going to go ahead and merge this in. If you want I can revisit and try to refactor again, but I'm assuming it's fine given your original comment 😬